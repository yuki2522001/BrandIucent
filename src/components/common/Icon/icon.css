@font-face {
  font-family: 'icomoon';
  src:  url(../../../assets/fonts/icomoon.eot?8d3ob1);
  src:  url(../../../assets/fonts/icomoon.eot?8d3ob1#iefix) format('embedded-opentype'),
    url(../../../assets/fonts/icomoon.ttf?8d3ob1) format('truetype'),
    url(../../../assets/fonts/icomoon.woff?8d3ob1) format('woff'),
    url(../../../assets/fonts/icomoon.svg?8d3ob1#icomoon) format('svg');
  font-weight: normal;
  font-style: normal;
  font-display: block;
}

[class^="icon-"], [class*=" icon-"] {
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'icomoon' !important;
  speak: never;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;

  /* Better Font Rendering =========== */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.icon-instagram:before {
  content: "\e901";
  color: var(--light);
}
.icon-twitter:before {
  content: "\e902";
  color: var(--light);
}
.icon-fb:before {
  content: "\e903";
  color: var(--light);
}
.icon-filter:before {
  content: "\e904";
  color: var(--light);
}
.icon-plus:before {
  content: "\e905";
  color: var(--text-primary);
}
.icon-cart:before {
  content: "\e906";
  color: var(--text-primary);
  font-size: 40px;
}
.icon-minus:before {
  content: "\e907";
  color: var(--text-primary);
}