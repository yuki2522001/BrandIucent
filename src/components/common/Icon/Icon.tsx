import React, { memo, MouseEventHandler } from "react";
import "./icon.css";

interface IconProps {
  iconName: string;
  onClick?: MouseEventHandler<HTMLElement>;
}

const Icon: React.FC<IconProps> = memo(({ iconName, onClick }) => {
  let className = "";
  switch (iconName) {
    case "filter":
      className = "icon-filter";
      break;
    case "plus":
      className = "icon-plus";
      break;
    case "minus":
      className = "icon-minus";
      break;
    case "cart":
      className = "icon-cart";
      break;
    case "fb":
      className = "icon-fb";
      break;
    case "twitter":
      className = "icon-twitter";
      break;
    case "instagram":
      className = "icon-instagram";
      break;
    default:
      break;
  }
  return <i data-testid="icon" onClick={onClick} className={className} />;
});

export default Icon;
