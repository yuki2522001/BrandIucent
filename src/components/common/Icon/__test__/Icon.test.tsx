import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Icon from "../Icon";

describe("Icon component", () => {
  test("should render component Icon type 'plus' ", () => {
    const { getByTestId } = render(<Icon iconName="plus" />);
    expect(getByTestId("icon")).toBeInTheDocument();
  });

  test("should render component Icon type 'filter' ", () => {
    const { getByTestId } = render(<Icon iconName="filter" />);
    expect(getByTestId("icon")).toBeInTheDocument();
  });

  test("should render component Icon type 'minus' ", () => {
    const { getByTestId } = render(<Icon iconName="minus" />);
    expect(getByTestId("icon")).toBeInTheDocument();
  });

  test("should render component Icon type 'cart' ", () => {
    const { getByTestId } = render(<Icon iconName="cart" />);
    expect(getByTestId("icon")).toBeInTheDocument();
  });

  test("should render component Icon type 'fb' ", () => {
    const { getByTestId } = render(<Icon iconName="fb" />);
    expect(getByTestId("icon")).toBeInTheDocument();
  });

  test("should render component Icon type 'twitter' ", () => {
    const { getByTestId } = render(<Icon iconName="twitter" />);
    expect(getByTestId("icon")).toBeInTheDocument();
  });

  test("should render component Icon type 'instagram' ", () => {
    const { getByTestId } = render(<Icon iconName="instagram" />);
    expect(getByTestId("icon")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Icon iconName="plus" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
