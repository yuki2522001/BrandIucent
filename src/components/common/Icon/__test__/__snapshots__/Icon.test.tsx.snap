// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component matches snapshot 1`] = `
<DocumentFragment>
  <i
    class="icon-plus"
    data-testid="icon"
  />
</DocumentFragment>
`;
